apps/explorer/src/utils/getAmount.ts

```diff
@@ -28,9 +28,18 @@ const getCoinType = (
     const events = txEffects?.events || [];
     const coinType = events
         ?.map((event: SuiEvent) => {
-            const data = Object.values(event).find(
-                (itm) => itm?.owner?.AddressOwner === address
-            );
+            const data = Object.values(event).find((itm) => {
+                if (
+                    itm?.changeType &&
+                    ['Receive', 'Pay'].includes(itm?.changeType) &&
```

Comment:
nit: we shouldn't need optional chaining on `itm.changeType` since we're checking that it is truthy on line 33. Same goes for `itm?.transactionModule`